pkg/job/discovery_test.go

```diff
@@ -454,3 +457,105 @@ func Test_getFilteredMetricDatas(t *testing.T) {
 		})
 	}
 }
+
+func getSampleMetricDatas(id string) *model.CloudwatchData {
+	return &model.CloudwatchData{
+		AccountID:              aws.String("123123123123"),
+		AddCloudwatchTimestamp: aws.Bool(false),
+		Dimensions: []*model.Dimension{
+			{
+				Name:  "FileSystemId",
+				Value: "fs-abc123",
+			},
+			{
+				Name:  "StorageClass",
+				Value: "Standard",
+			},
+		},
+		ID:        aws.String(id),
+		MetricID:  aws.String(id),
+		Metric:    aws.String("StorageBytes"),
+		Namespace: aws.String("efs"),
+		NilToZero: aws.Bool(false),
+		Period:    60,
+		Region:    aws.String("us-east-1"),
+		Statistics: []string{
+			"Average",
+		},
+		Tags: []model.Tag{
+			{
+				Key:   "Value1",
+				Value: "",
+			},
+			{
+				Key:   "Value2",
+				Value: "",
+			},
+		},
+	}
+}
+
+func BenchmarkMapResultsToMetricDatas(b *testing.B) {
+	type testcase struct {
+		metricsPerQuery    int
+		testResourcesCount int
+		metricsPerResource int
+	}
+
+	for name, tc := range map[string]testcase{
+		"small case": {
+			metricsPerQuery:    50,
+			testResourcesCount: 10,
+			metricsPerResource: 100,
+		},
+		"big case": {
+			metricsPerQuery:    50,
+			testResourcesCount: 100,
+			metricsPerResource: 1000,
+		},
```

Comment:
```suggestion
	for name, tc := range map[string]testcase{
		"small case": {
			metricsPerQuery:    500,
			testResourcesCount: 10,
			metricsPerResource: 10,
		},
		"medium case": {
			metricsPerQuery:    500,
			testResourcesCount: 1000,
			metricsPerResource: 50,
		},
		"big case": {
			metricsPerQuery:    500,
			testResourcesCount: 2000,
			metricsPerResource: 50,
		},
```

I wonder if these number might be a little closer to real world use cases. `GetMetricData` can at most take 500 metrics per query and I think most services tend to have < 50 metrics.